cmd/nex/node.go

```diff
@@ -0,0 +1,317 @@
+	WorkloadTypes                    WorkloadConfigs   `help:"Workload types configurations for nex node to initialize"`
+
+	HostServicesConfig HostServicesConfig `embed:"" prefix:"hostservices." group:"Host Services Configuration"`
+	OtelConfig         OtelConfig         `embed:"" prefix:"otel." group:"OpenTelemetry Configuration"`
+}
+
+func (u Up) Validate() error {
+	var errs error
+	if u.WorkloadTypes == nil || len(u.WorkloadTypes) < 1 {
+		errs = errors.Join(errs, errors.New("attempting to start nex node with no workload types configured. Please provide at least 1 workload type configuration"))
+	}
+	return errs
+}
+
+func (u Up) Run(ctx context.Context, globals Globals, n *Node) error {
+	if globals.Check {
+		return printTable("Node Up Configuration", append(globals.Table(), u.Table()...)...)
+	}
+
+	nc, err := configureNatsConnection(globals)
+	if err != nil {
+		return err
+	}
+
+	kp, err := nkeys.CreateServer()
+	if err != nil {
+		return err
+	}
+
+	pubKey, err := kp.PublicKey()
+	if err != nil {
+		return err
+	}
+
+	logger := configureLogger(globals, nc, pubKey)
+
+	nexNode, err := node.NewNexNode(nc,
+		node.WithLogger(logger),
+		node.WithAgentHandshakeTimeout(u.AgentHandshakeTimeoutMillisecond),
+		node.WithResourceDirectory(u.DefaultResourceDir),
+		node.WithInternalNodeNATHost(u.InternalNodeHost, u.InternalNodePort),
+		node.WithNodeTags(u.Tags),
+		node.WithValidIssuers(u.ValidIssuers),
+		node.WithOTelOptions(node.OTelOptions{
+			MetricsEnabled:   u.OtelConfig.OtelMetrics,
+			MetricsPort:      u.OtelConfig.OtelMetricsPort,
+			MetricsExporter:  u.OtelConfig.OtelMetricsExporter,
+			TracesEnabled:    u.OtelConfig.OtelTraces,
+			TracesExporter:   u.OtelConfig.OtelTracesExporter,
+			ExporterEndpoint: u.OtelConfig.OtlpExporterUrl,
+		}),
+		node.WithWorkloadTypes(func() []node.WorkloadOptions {
+			ret := make([]node.WorkloadOptions, len(u.WorkloadTypes))
+			for i, e := range u.WorkloadTypes {
+				ret[i] = node.WorkloadOptions{
+					Name:     e.Name,
+					AgentUri: e.AgentUri,
+					Argv:     e.Argv,
+					Env:      e.Env,
+				}
+			}
+			return ret
+		}()),
+		node.WithHostServiceOptions(func() node.HostServiceOptions {
+			return node.HostServiceOptions{
+				NatsUrl:      u.HostServicesConfig.NatsUrl,
+				NatsUserJwt:  u.HostServicesConfig.NatsUserJwt,
+				NatsUserSeed: u.HostServicesConfig.NatsUserSeed,
+				Services: func() map[string]node.ServiceConfig {
+					ret := make(map[string]node.ServiceConfig, len(u.HostServicesConfig.Services))
+					for k, v := range u.HostServicesConfig.Services {
+						ret[k] = node.ServiceConfig{
+							Enabled:       v.Enabled,
+							Configuration: v.Configuration,
+						}
+					}
+					return ret
+				}(),
+			}
+		}()),
+	)
+	if err != nil {
+		return err
+	}
+
+	err = nexNode.Validate()
+	if err != nil {
+		return err
+	}
+
+	logger.Info("Validating Nex Node")
+	err = nexNode.Validate()
+	if err != nil {
+		return err
+	}
+
+	logger.Info("Starting Nex Node")
+	nexNode.Start()
```

Comment:
might be worth leaving a comment here that this code requires `Start` to be blocking